src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts

```diff
@@ -46,4 +149,44 @@ export class PowerPagesChatParticipant {
 
     };
 
+    private async intializeOrgDetails(): Promise<void> {
+
+        if (this.isOrgDetailsInitialized) {
+            return;
+        }
+
+        this.isOrgDetailsInitialized = true;
+
+        const orgDetails: OrgDetails | undefined = this.extensionContext.globalState.get('orgDetails');
+
+        if (orgDetails) {
+            this.orgID = orgDetails.orgID;
+            this.orgUrl = orgDetails.orgUrl;
+        } else {
+            if (this._pacWrapper) {
+                const pacActiveOrg = await this._pacWrapper.activeOrg();
+                if (pacActiveOrg && pacActiveOrg.Status === PAC_SUCCESS) {
+                    this.handleOrgChangeSuccess(pacActiveOrg.Results);
+                } else {
+                    await createAuthProfileExp(this._pacWrapper);
+                }
+            }
+        }
+    }
+
+    private async handleOrgChangeSuccess(orgDetails: ActiveOrgOutput) {
+        this.orgID = orgDetails.OrgId;
+        this.orgUrl = orgDetails.OrgUrl
+
+        this.extensionContext.globalState.update('orgDetails', { orgID: this.orgID, orgUrl: this.orgUrl });
+
+        //TODO: Handle AIB GEOs
+    }
+
+    async getEndpoint(orgID: string, telemetry: ITelemetry) {
+        if (!this.cachedEndpoint) {
+            this.cachedEndpoint = await getIntelligenceEndpoint(orgID, telemetry, '') as { intelligenceEndpoint: string; geoName: string };
+        }
+        return this.cachedEndpoint;
+    }
```

Comment:
can both go in their own utils file - do not over-grow this file - this should only have content specific to working of the agent